src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp

```diff
@@ -66,27 +505,397 @@ class boss_thorim : public CreatureScript
                     Talk(SAY_SLAY);
             }
 
-            void JustDied(Unit* /*killer*/) override
+            void SpellHit(Unit* /*caster*/, SpellInfo const* spellInfo) override
             {
-                Talk(SAY_DEATH);
+                if (spellInfo->Id == SPELL_TOUCH_OF_DOMINION_TRIGGERED)
+                {
+                    if (Creature* sif = instance->GetCreature(DATA_SIF))
+                    {
+                        sif->AI()->Talk(SAY_SIF_DESPAWN);
+                        sif->DespawnOrUnsummon(6000);
+                        _hardMode = false;
+                    }
+                }
+            }
+
+            void SpellHitTarget(Unit* who, SpellInfo const* spellInfo) override
+            {
+                if (who->GetTypeId() == TYPEID_PLAYER && spellInfo->Id == SPELL_LIGHTNING_RELEASE)
+                    _dontStandInTheLightning = false;
+            }
+
+            void FinishEncounter()
+            {
+                if (_encounterFinished)
+                    return;
+
+                _encounterFinished = true;
+
+                DoCastAOE(SPELL_CREDIT_KILL, true); // before change faction
+
+                me->SetReactState(REACT_PASSIVE);
+                me->InterruptNonMeleeSpells(true);
+                me->RemoveAllAttackers();
+                me->AttackStop();
```

Comment:
It's different. We mustn't stop combat, rather keep on combat until the outro finishes